ofrak_core/ofrak/core/elf/analyzer.py

```diff
@@ -322,15 +322,11 @@ async def analyze(self, resource: Resource, config=None) -> AttributesType[Named
         elf_r = await section.get_elf()
         string_section = await elf_r.get_section_name_string_section()
         try:
-            string_section_data = await string_section.resource.get_data(
-                Range(section_header.sh_name, Range.MAX)
+            ((_, raw_section_name),) = await string_section.resource.search_data(
+                re.compile(b".[^\x00]+\x00"), start=section_header.sh_name, max_matches=1
```

Comment:
It probably makes sense to compile this outside of this function call. Also, why does there have to be a non-newline character, followed by one or more non-null bytes? I think the initial `.` is erroneous.

```suggestion
                re.compile(b"[^\x00]+\x00"), start=section_header.sh_name, max_matches=1
```